equinox-shipment/Domain/FinalizationTransaction.fs

```diff
@@ -0,0 +1,90 @@
+module FinalizationTransaction
+
+module Events =
+
+    let [<Literal>] CategoryId = "FinalizationTransaction"
+
+    let (|ForClientId|) (clientId: string) = FsCodec.StreamName.create CategoryId clientId
+
+    type Event =
+        | FinalizationRequested of containerId: string * shipmentIds: string[]
+
+        | AssignmentCompleted   of containerId: string * shipmentIds: string[]
+        | FinalizationCompleted of containerId: string
+
+        | RevertRequested       of containerId: string * shipmentIds: string[]
+        | FinalizationFailed    of containerId: string
+        interface TypeShape.UnionContract.IUnionContract
+
+    let codec = FsCodec.NewtonsoftJson.Codec.Create<Event>()
+
+type Action =
+    | AssignShipments   of containerId: string * shipmentIds: string[]
+    | FinalizeContainer of containerId: string * shipmentIds: string[]
+    // Reverts the assignment of the container for the shipments provided.
+    | RevertAssignment  of containerId: string * shipmentIds: string[]
+    | Finish of bool
```

Comment:
`Finish of success : bool` + maybe Exit or Complete is better name?

Reply:
Updated thanks!

I did not add an exit code as for this example I don't know what exit code to put.